lib/charms/layer/ctl/__base__.py

```diff
@@ -0,0 +1,38 @@
+class ContainerRuntimeCtlBase(object):
+    """
+    Base class for Container Runtime ctls.
+    """
+    def __init__(self):
+        """
+        :return: None
+        """
+        pass
+
+    def _exec(self):
+        """
+        Call the underlying CLI.
+
+        :return: String output
+        """
+        raise NotImplementedError
+
+    def run(self, name, image, command=None, *args):
```

Comment:
How do I pass in environment variables and volume mounts to the container? How do I set privileged or non-privileged? It's not very clear from looking at this.

Reply:
@Cynerva I need to populate a list of all the settings we need to pass in through these methods; this is currently just an example.  We will have to make them all named arguments to then translate them to the relevant CLI.